### PR TITLE
GTEST/UCP: Skipped AM datatype test for ud_v and ud_x.

### PR DESCRIPTION
**Pull request by rakhmets**
## What
Skip `test_ucp_am_nbx_dts.short_bcopy_send` and `test_ucp_am_nbx_dts.zcopy_send` for UD transports in new proto infrastructure.

## Why
There are unstable reproducible failures during `roce on worker` jobs. The failures look like:
```
[ RUN      ] ud/test_ucp_am_nbx_dts.short_bcopy_send/0 <ud_v>
[1689063810.082094] [swx-rain04:448254:0]        ud_verbs.c:255  UCX  ERROR Invalid am_bcopy length: 5944 (expected: <= 4096)
[1689063820.115899] [swx-rain04:448254:0]       ucp_test.cc:280  UCX  ERROR operation returned error: Invalid parameter
```
```
[ RUN      ] udx/test_ucp_am_nbx_dts.zcopy_send/0 <ud_x>
[1689075622.531834] [swx-rain04:334703:0]         ud_mlx5.c:269  UCX  ERROR Invalid am_zcopy payload length: 5944 (expected: <= 4096)
[1689075632.566755] [swx-rain04:334703:0]       ucp_test.cc:280  UCX  ERROR operation returned error: Invalid parameter
```

### CODE REVIEW

File: test/gtest/ucp/test_ucp_am.cc [hunk 1]
```diff
@@ -879,17 +881,14 @@ UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx)
...
-
 protected:
     virtual unsigned get_send_flag() const
     {
         return UCP_AM_SEND_FLAG_REPLY;
     }
 };
 
-bool test_ucp_am_nbx_reply_always::am_lane_has_caps(uint64_t caps)
+bool test_ucp_am_nbx::am_lane_has_caps(uint64_t caps)
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:891]: move this func up

File: test/gtest/ucp/test_ucp_am.cc [hunk 2]
```diff
@@ -1510,8 +1509,10 @@ class test_ucp_am_nbx_dts : public test_ucp_am_nbx_reply {
     }
 };
 
-UCS_TEST_P(test_ucp_am_nbx_dts, short_bcopy_send, "ZCOPY_THRESH=-1",
-           "RNDV_THRESH=-1")
+UCS_TEST_SKIP_COND_P(test_ucp_am_nbx_dts, short_bcopy_send,
+                     !is_proto_disabled() &&
+                             has_any_transport({"ud_v", "ud_x"}),
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1514]: add comment why

File: test/gtest/ucp/test_ucp_am.cc [hunk 3]
```diff
@@ -1520,8 +1521,15 @@ UCS_TEST_P(test_ucp_am_nbx_dts, short_bcopy_send, "ZCOPY_THRESH=-1",
...
-UCS_TEST_P(test_ucp_am_nbx_dts, zcopy_send, "ZCOPY_THRESH=1", "RNDV_THRESH=-1")
+UCS_TEST_SKIP_COND_P(test_ucp_am_nbx_dts, zcopy_send,
+                     !is_proto_disabled() &&
+                             has_any_transport({"ud_v", "ud_x"}),
+                     "ZCOPY_THRESH=1", "RNDV_THRESH=-1")
 {
+    if (!am_lane_has_caps(UCT_IFACE_FLAG_AM_ZCOPY)) {
+        UCS_TEST_SKIP_R("am_zcopy is not supported");
+    }
+
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1532]: move to another pr

File: test/gtest/ucp/test_ucp_am.cc (hunk 2 above)
Comment (rakhmets) [reply test/gtest/ucp/test_ucp_am.cc:1517]: Added

File: test/gtest/ucp/test_ucp_am.cc (hunk 3 above)
Comment (rakhmets) [reply test/gtest/ucp/test_ucp_am.cc]: Moved to https://github.com/openucx/ucx/pull/9215

File: test/gtest/ucp/test_ucp_am.cc (hunk 1 above)
Comment (rakhmets) [reply test/gtest/ucp/test_ucp_am.cc]: Fixed in https://github.com/openucx/ucx/pull/9215

**Review by yosefe — APPROVED** [review]
pls squash

**Review by yosefe — APPROVED** [review]
